storage/file_persister.go

```diff
@@ -13,14 +13,91 @@ import (
 	"path/filepath"
 	"strings"
 	"time"
+
+	"github.com/grafana/xk6-browser/env"
 )
 
+// FilePersister is the type that all file persisters must implement. It's job is
+// to persist a file somewhere, hiding the details of where and how from the caller.
+type FilePersister interface {
+	Persist(ctx context.Context, path string, data io.Reader) (err error)
+}
+
+// NewFilePersister will return either a LocalFilePersister or a RemoteFilePersister
+// depending on whether the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the
+// correct configs.
+func NewFilePersister(envLookup env.LookupFunc) (FilePersister, error) {
+	envVar, ok := envLookup(env.ScreenshotsOutput)
+	if !ok || envVar == "" {
+		return &LocalFilePersister{}, nil
+	}
+
+	u, b, h, err := parseEnvVar(envVar)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %s: %w", env.ScreenshotsOutput, err)
+	}
+
+	return NewRemoteFilePersister(u, h, b), nil
+}
+
+// parseEnvVar will parse a value such as:
+// url=https://127.0.0.1/,basePath=/screenshots,header.1=a,header.2=b
+// and return them.
+func parseEnvVar(envVarValue string) (string, string, map[string]string, error) {
```

Comment:
This way of parsing an env var seems super flakey, am i missing a trick?

Reply:
Not that I know of :( Maybe `strings.Cut` can help here.

Reply:
> Not that I know of :( Maybe strings.Cut can help here

No worries 👍 

> BTW, it might be a better idea to return this in a struct instead of a lot of primitive values.

I was wondering whether it would be better to do that instead of returning a long list of variables. I'll change it to return a struct.

Reply:
I believe a struct will be much clearer. We might also want to make these field names descriptive or add some comments.

Reply:
I've added this change in 3389a93947d654e531f932431213a1ba80240c5e.

Reply:
Better 👍 For this:

https://github.com/grafana/xk6-browser/blob/3389a93947d654e531f932431213a1ba80240c5e/browser/file_persister.go#L11-L15

WDYT about `presignedURLConfig` since there isn't another remote persisted config, and it's about pre-signed URLs. This also simplifies the field names. Plus, this is used in a very short scope—its context makes it already meaningful.

```go
type presignedURLConfig struct {
	getterURL string
	headers   map[string]string
	basePath  string
}
```

No strong opinions.

Reply:
Makes sense, changed in d864d6d8058d44e6da07a172014e950fcbbc21a2

Reply:
Nice. Thanks.